### PR TITLE
(chore) Bump framework and update patient banner props

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary
This PR bumps the framework and openmrs tooling, and updates the patient banner props following the changes in https://github.com/openmrs/openmrs-esm-core/pull/1158

## Screenshots
None

## Related Issue
None

### CODE REVIEW

**Review by denniskigen — APPROVED** [review]
Thanks for working on this, @vasharma05. I've taken the liberty of pushing a commit that, among other small tweaks:

- Also updates the version of the Common Lib dependency (in the hopes of fixing [this issue](https://openmrs.slack.com/archives/CHP5QAE5R/p1728484317351759))
- Switching from using `String(Math.random())` to generate random UUIDs for the `name` and `address` properties of the FHIR patient object to using `uuidv4` which actually generates unique v4 UUIDs.

File: packages/esm-patient-search-app/src/compact-patient-search/compact-patient-banner.component.tsx [hunk 1]
```diff
@@ -1,3 +1,7 @@
+import React, { forwardRef, useContext, useMemo } from 'react';
+import classNames from 'classnames';
+import { v4 as uuidv4 } from 'uuid';
+import { useTranslation } from 'react-i18next';
```
Comment (denniskigen) [inline packages/esm-patient-search-app/src/compact-patient-search/compact-patient-banner.component.tsx:4]: This change just reorganises the imports (note to self to prioritise work on adding https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md to our ESLint config for automatically handling this concern)

File: packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.component.tsx [hunk 2]
```diff
@@ -132,11 +133,6 @@ const PatientBanner: React.FC<PatientBannerProps> = ({ patient, patientUuid, hid
   );
 };
 
-interface ClickablePatientContainerProps {
-  patientUuid: string;
-  children: React.ReactNode;
-}
```
Comment (denniskigen) [inline packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.component.tsx:138]: I've just lifted this interface higher up in the file.

File: packages/esm-patient-search-app/src/compact-patient-search/compact-patient-banner.component.tsx (hunk 1 above)
Comment (vasharma05) [reply packages/esm-patient-search-app/src/compact-patient-search/compact-patient-banner.component.tsx:4]: That's amazing!

**Pull request merged**